DifferentiationInterface/ext/DifferentiationInterfaceMooncakeExt/twoarg.jl

```diff
@@ -27,43 +39,36 @@ function DI.value_and_pullback(
     contexts::Vararg{Context,C},
 ) where {C}
     dy = only(ty)
-    dy_righttype = convert(tangent_type(typeof(y)), copy(dy))
+
+    # Set all tangent storage to zero.
+    df! = set_to_zero!!(prep.df!)
+    # dx_righttype = set_to_zero!!(prep.dx_righttype)  # TODO: why doesn't this work?
```

Comment:
For some reason when I use this line instead of the following line, the tests fail. They only fail for batch mode with 2 tangents, so my guess is that something is not being properly zero-ed/reset. Any clue?

Reply:
When you say an error, is it a numerical error, or does something fail to run?

Reply:
It's a numerical error. Things run properly but presumably the pullback for the second tangent is wrong, whereas the pullback for the first tangent was right. So somewhere, something fails to reset between both calls

Reply:
Hmmm yeah, that is weird.

Reply:
Ah, I think I see what's going on. We directly return `dx_righttype`, rather than writing it out to some other buffer. Since `prep.dx_righttype` and `tangent(fdata(dx_righttype), new_dx)` can alias each other, there's memory sharing going on between each iteration.

I would imagine the the results you're seeing if you request the gradients in batch mode is that they will all be the same. Moreover, assuming you're iterating through the tangents in order, they'll all have the correct gradient for the last tangent in the batch. Does that make sense?

Reply:
Oh yes, classic mistake of returning whatever is in the preparation object instead of making a copy. I'll fix it and merge

Reply:
Do you have the option to do prepration which is specific to batch mode? If so, you could pre-allocate buffers for all of the results.

Reply:
Yes, I have opened https://github.com/gdalle/DifferentiationInterface.jl/issues/518 to keep track